### PR TITLE
[processing] add function to get processing version

### PR DESCRIPTION
**Pull request by Gustry**
As processing changes its API sometimes, we need to know the current processing version in our plugin.

CC : @volaya @alexbruy

### CODE REVIEW

File: python/plugins/processing/tools/general.py [hunk 1]
```diff
@@ -75,3 +76,26 @@ def runalg(algOrName, *args, **kwargs):
...
+
+def version():
+    """Returns the Processing version number.
+
+    :returns: The version number.
+    :rtype: int
+    """
+    # Get location of application wide version info
+    root_dir = abspath(join(dirname(__file__), '..'))
+    fid = open(join(root_dir, 'metadata.txt'))
```
Comment (m-kuhn) [inline python/plugins/processing/tools/general.py]: In general it's preferable to use

with open(filename) as file:
    filed.readlines() # or something else

it's shorter and handles exceptions etc.

**Comment by m-kuhn** [comment]
Could you insert a call to print into the algorithm tests, so we have this in the context of test output?

https://github.com/qgis/QGIS/blob/master/python/plugins/processing/tests/AlgorithmsTestBase.py#L53

**Comment by alexbruy** [comment]
Not sure if this can help a lot. Currently Processing version changed only when new QGIS version released. To make this useful we need to change Processing version almost on every commit which touches its code.

Maybe better to accept Tim's suggestion and just postpone all breaking changes until major QGIS version is released.

Regarding PR itself, I just commited improved code.

**Comment by Gustry** [comment]
According to https://plugins.qgis.org/plugins/processing/, the plugin seems not to be updated exactly at each QGIS release.
Moreover, some users may not update their plugins every time, because they don't do it or because their are running two versions of QGIS (2.8 and latest) and the latest Processing is not compatible with QGIS 2.8, so I usually let an old version of Processing.
As a QGIS user, it's fine, I can use whatever processing version, but in our plugin, we need to be sure which Processing version the user has.

> Regarding PR itself, I just commited improved code.
So the code is in master ? I didn't see any commit.

**Comment by gioman** [comment]
> but in our plugin, we need to be sure which Processing version the user has.

agree, this is useful especially now that Processing shipped in QGIS 2.14 changed in the Processing plugins is needed to check that Processing version used is not an old one (installed as a plugin update).

**Pull request closed without merging**